File: problems/s/shatteredcake/shatteredcake.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int w, n, wi, li, area;

    cin >> w;
    cin >> n;

    area = 0;
    for (int i=0; i<n; i++) {
        cin >> wi >> li;
        area +=  wi * li;
    }

    cout << area/w << endl;
}

File: problems/s/shatteredcake/shatteredcake.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    w = int(input())
    n = int(input())

    area = 0
    for _ in range(n):
        wi, li = [int(val) for val in input().split()]
        area += wi * li
    print(area//w)


if __name__ == '__main__':
    main()